components/org.wso2.carbon.consent.mgt.ui/src/main/resources/web/consent/add-purpose.jsp

```diff
@@ -99,9 +121,30 @@
         function doFinish() {
             document.dataForm.action = "<%=Encode.forJavaScript(addFinishPurposePage)%>";
             if (doValidation() === true) {
+                if(!doValidationForMandatoryEmailP2() && !<%=hasPurposeWithMandatoryEmailInList%> ) {
+                    CARBON.showWarningDialog("<%=resourceBundle.getString("missing.mandatory.email.pii.category.warning.add.purpose")%>",
+                    doSubmit,doSubmit);
+                }else {
+                    doSubmit();
+                }
+            }
+            function doSubmit() {
                 document.dataForm.submit();
             }
         }
+        function doValidationForMandatoryEmailP2() {
```

Comment:
what is P2 at the end of the function name?

Reply:
PII category

Reply:
Changed as doValidationForMandatoryEmailPIICategory